src/Serializer/AbstractItemNormalizer.php

```diff
@@ -231,7 +231,13 @@ public function denormalize($data, $class, $format = null, array $context = [])
         $context['api_denormalize'] = true;
         $context['resource_class'] = $resourceClass;
 
-        if (null !== ($inputClass = $this->getInputClass($resourceClass, $context)) && null !== ($dataTransformer = $this->getDataTransformer($data, $resourceClass, $context))) {
+        $inputClass = $this->getInputClass($resourceClass, $context);
+        $dataTransformer = $this->getDataTransformer($data, $resourceClass, $context);
```

Comment:
No need to call this function if `$inputClass` is `null`. This matters because we're on the hot path.